src/perform_step/srk_weak.jl

```diff
@@ -152,8 +152,32 @@
       end
   end
 
-  # Test EM scheme:
-  # integrator.u = uprev + 1*k1*dt + 1*g1*_dW
+  if integrator.opts.adaptive && (typeof(W.dW) <: Number || is_diagonal_noise(integrator.sol.prob) || m==1)
+
+    # schemes with lower convergence order
+    if c03!=0.0
+      # scheme has det. conv. order 3 and we look for det. conv. order 2 scheme
+      rat = c02/c03
+      δ₁ = integrator.opts.delta*(rat-1)
+      δ₂ = -integrator.opts.delta*rat
+      uhat = uprev + ((α1+δ₁)*k1 + (α2+integrator.opts.delta)*k2 + (α3+δ₂)*k3)*dt
+    else
+      # check against EM
+      uhat = uprev + k1*dt
+    end
+    if is_diagonal_noise(integrator.sol.prob)
+      uhat += g1 .* _dW
+    else
+      uhat += g1 * _dW
+    end
+
+    Eprev = Statistics.mean(uprev,dims=2)
```

Comment:
I don't think you need to take a mean on these, since it will naturally happen as part of the norm on the resids

Reply:
it won't actly be a mean, but it'll be an average of sorts

Reply:
oh yes.. I just checked that they are indeed quite close by printing the values for the Brusselator example. .. at first I simply checked it with `isapprox()` (which is false.. therefore I went initially for the version with the mean to be as close as possible to Roesslers description.)